learn/work/sparse-recipe/index.qmd

```diff
@@ -0,0 +1,254 @@
+---
+title: "Model tuning using a recipe"
+categories:
+ - tuning
+ - classification
+ - sparse data
+type: learn-subsection
+weight: 1
+description: | 
+ Fitting a model using tidymodels where sparse data is created using a recipe.
+toc: true
+toc-depth: 2
+include-after-body: ../../../resources.html
+---
+
+```{r}
+#| label: "setup"
+#| include: false
+#| message: false
+#| warning: false
+source(here::here("common.R"))
+```
+
+  
+```{r}
+#| label: "load"
+#| include: false
+library(tidymodels)
+library(nycflights13)
+set.seed(1234)
+
+pkgs <- c("tidymodels", "nycflights13")
+
+theme_set(theme_bw() + theme(legend.position = "top"))
+```
+
+## Introduction
+
+`r article_req_pkgs(pkgs)`
+
+
+This article demonstrates how we can use a recipe to generate data sparsity in tidymodels.
+
+We use the term **sparse data** to denote a data set that contains a lot of 0s. Such data is commonly seen as a result of dealing with categorical variables, text tokenization, or graph data sets. The word sparse describes how the information is packed. Namely, it represents the presence of a lot of zeroes. For some tasks, we can easily get above 99% percent of 0s in the predictors. 
+
+The reason we use sparse data as a construct is that it is a lot more memory efficient to store the positions and values of the non-zero entries than to encode all the values. One could think of this as a compression, but one that is done such that data tasks are still fast. The following vector requires 25 values to store it normally (dense representation). This representation will be referred to as a **dense vector**.
+
+```r
+c(100, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0)
+```
+The sparse representation of this vector only requires 5 values. 1 value for the length (25), 2 values for the locations of the non-zero values (1, 22), and 2 values for the non-zero values (100, 1). This idea can also be extended to matrices as is done in the Matrix package.
+
+Not all recipes steps can handle sparsity, we have a [list of compatible](../../../find/sparse/index.qmd) steps you can use to guide the recipe creation.
+
+## The data
+
+We will be using the [nycflights13](https://nycflights13.tidyverse.org/) data set for this demonstration. We are using this data specifically because it contains a number of categorical with a lot of levels, that when dummified will create a lot of sparse columns.
```

Comment:
```suggestion
We will be using the [nycflights13](https://nycflights13.tidyverse.org/) data set for this demonstration. We are using this data specifically because it contains a number of categorical with a lot of levels, that when converted to binary indicator columns (a.k.a. "dummy variables") will create a lot of sparse columns.
```